src/dmd/toobj.d

```diff
@@ -806,7 +803,7 @@ void toObjFile(Dsymbol ds, bool multiobj)
                     }
                     else if (auto f = s.isFuncDeclaration())
                     {
-                        objmod.setModuleCtorDtor(s.csym, isCtor);
+                        f.isCrtCtorDtor |= isCtor ? 1 : 2;
```

Comment:
Why the `|=` ? Is there ever a case where 1 and 2 can both be set? Or any other bits?

Reply:
I was a bit surprised, too, but the existing implementation allows both: https://github.com/dlang/dmd/blob/master/test/runnable/test17868b.d#L7

Reply:
The pragma just places a function pointer in the init/fini segment, so as a low-level tool it can be used as people want.